### ISSUE TITLE
Menu not setting focus to link in first menu item containig link when navigating with keyboard

### ISSUE THREAD

**Issue by kpustakhod** [ant-design/ant-design#57766]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/vuy6tohz?file=src%2Fdemo.tsx)

### Steps to reproduce

1. Create a DropdownButton component with menu where all items are links (check code block below).
2. Add autoFocus prop to DrodownButton (this ctep can be skipped, otherwise you just need to focus to menu manually with Tab key).
3. Focus with keyboard on DropdownButton, press enter.
4. Menu opens, focus on menu (if not using autoFocus)
5. Fist menu item is focused. Press Enter.

```
import React from 'react';

import type { MenuProps } from 'antd';
import { Button, Dropdown, Space } from 'antd';

const items: MenuProps['items'] = [
  {
    key: '1',
    label: (
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://www.antgroup.com"
      >
        1st menu item
      </a>
    ),
  },
  {
    key: '2',
    label: (
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://www.aliyun.com"
      >
        2nd menu item
      </a>
    ),
  },
  {
    key: '3',
    label: (
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://www.luohanacademy.com"
      >
        3rd menu item
      </a>
    ),
  },
];

const App: React.FC = () => (
  <Space vertical>
    <Space wrap>
      <Dropdown trigger={['click']} menu={{ items }}>
        <Button>Menu with links</Button>
      </Dropdown>
    </Space>
  </Space>
);

export default App;
```

### What is expected?

User should be navigated to url from html link.

### What is actually happening?

Menu closes, navigation not happening.

| Environment | Info |
| --- | --- |
| antd | 6.3.6 |
| React | React 19 |
| System | iOS 26.3.1 (a) |
| Browser | Chrome 146.0.7680.80 |

---
Currently focus is set on menu item container, and if menu item contains link, it is not possible to follow by link url with keyboard.

**Comment by afc163** [comment]
Thanks for the report. This is a valid accessibility issue — when a Dropdown/Menu opens via keyboard, the focus should be set on the first menu item's link element so users can immediately press Enter to activate it, rather than only focusing the menu container.

Would you like to submit a PR to fix this?
